extra_data/components.py

```diff
@@ -2013,7 +2013,10 @@ def identify_multimod_detectors(
     res = set()
     for cls in clses:
         for name in cls._find_detector_names(data):
-            res.add((name, cls))
+            if (detector_name is None
+                or detector_name in name
```

Comment:
I think if you're specifying a name, it should be exact; this matches how  the`detector_name` is [used when instatiating a component](https://github.com/European-XFEL/EXtra-data/blob/c6307c5fad46aa8a751461745085f81adba3237d/extra_data/components.py#L269).

```suggestion
                or detector_name == name
```

Reply:
Thanks Thomas, looks good to me. James, would you like to commit this?